network/phonebook.go

```diff
@@ -256,3 +340,16 @@ func (mp *MultiPhonebook) UpdateRetryAfter(addr string, retryAfter time.Time) {
 		op.UpdateRetryAfter(addr, retryAfter)
 	}
 }
+
+// WaitAndAddConnectionTime will wait to prevent exceeding connectionsRateLimitingCount.
+// Then it will register the next connection time.
+func (mp *MultiPhonebook) WaitAndAddConnectionTime(addr string,
+	connectionsRateLimitingCount uint, connectionsRateLimitingWindowSeconds uint) {
+	mp.lock.Lock()
+	defer mp.lock.Unlock()
+	for _, op := range mp.phonebookMap {
+		op.WaitAndAddConnectionTime(addr,
+			connectionsRateLimitingCount, connectionsRateLimitingWindowSeconds)
+		break // If the first phonebook does not have the phonebookData, will add it there
```

Comment:
iteration over maps aren't guaranteed to be executed at the same order.
my suggestion is that if we can't find the address on any of the maps we would just exit.